From 36df18c4f64ead85b6a9619364908fba736fdd1d Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 4 Jun 2020 16:08:09 -0700
Subject: [PATCH] Fix build with boost-1.73.0

* fixes:
umediaserver/1.0.0-24-r12/git/src/media_client/uMediaClient.cpp:68:8: error: 'replace_if' is not a member of 'std'
umediaserver/1.0.0-24-r12/git/src/media_client/uMediaClient.cpp:68:8: error: 'replace_if' is not a member of 'std'

and

In file included from umediaserver/1.0.0-24-r12/recipe-sysroot/usr/include/boost/bind.hpp:30,
                 from umediaserver/1.0.0-24-r12/recipe-sysroot/usr/include/boost/property_tree/json_parser/detail/parser.hpp:7,
                 from umediaserver/1.0.0-24-r12/recipe-sysroot/usr/include/boost/property_tree/json_parser/detail/read.hpp:13,
                 from umediaserver/1.0.0-24-r12/recipe-sysroot/usr/include/boost/property_tree/json_parser.hpp:16,
                 from umediaserver/1.0.0-24-r12/git/src/utils/rmc.cpp:37:
umediaserver/1.0.0-24-r12/recipe-sysroot/usr/include/boost/bind.hpp:36:1: note: '#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.'
   36 | BOOST_PRAGMA_MESSAGE(
      | ^~~~~~~~~~~~~~~~~~~~
---
 src/media_client/uMediaClient.cpp | 1 +
 src/process_pool/ProcessPool.cpp  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/media_client/uMediaClient.cpp b/src/media_client/uMediaClient.cpp
index e865c9b..cd19a18 100755
--- a/src/media_client/uMediaClient.cpp
+++ b/src/media_client/uMediaClient.cpp
@@ -25,6 +25,7 @@
 
 #include "uMediaClient.h"
 
+#include <algorithm>
 #include <UMSConnector.h>
 #include <Logger_macro.h>
 #include <stdlib.h>
diff --git a/src/process_pool/ProcessPool.cpp b/src/process_pool/ProcessPool.cpp
index e236f94..02b132f 100644
--- a/src/process_pool/ProcessPool.cpp
+++ b/src/process_pool/ProcessPool.cpp
@@ -120,7 +120,7 @@ void ProcessPool::start(const std::string & type) {
 	std::string cmd = config.bin + " -s" + service_name;
 	using namespace std::placeholders;
 	Process::ptr_t proc = std::make_shared<Process>
-			(cmd, std::bind(&ProcessPool::exited, this, _1 ,_2), config.env_vars);
+			(cmd, std::bind(&ProcessPool::exited, this, std::placeholders::_1 ,std::placeholders::_2), config.env_vars);
 	_started_pool.emplace(std::make_pair(service_name, proc_info_t{type, service_name, proc}));
 
 	LOG_INFO(_log, MSGNFO_PROC_STARTED, "New process %d of type %s started.", proc->pid(), type.c_str());
